socketshark/subscription.py

```diff
@@ -78,6 +81,38 @@ def prepare_service_data(self):
         return await self.perform_service_request('on_unsubscribe',
                                                   raise_error=False)
 
+    def filter_message(self, data):
+        """
+        Returns whether to filter the given message.
+        """
+        # Check whether the message is filtered by comparing any defined
+        # filter_fields to auth_info.
+        filter_fields = self.service_config.get('filter_fields', [])
```

Comment:
Separate to independent functions like `drop_based_on_field` and `drop_out_of_order`.